src/components/customSearch/CustomSearch.tsx

```diff
@@ -0,0 +1,225 @@
+                    borderRadius: '4px',
+                    boxShadow: '0px 0px 0px 1px #E0E0E0',
+                    mt: 4,
+                    padding: 2,
+                    display: 'flex'
+                  }}
+                >
+                  <Grid container direction="column" width="100%">
+                    <Grid
+                      item
+                      container
+                      direction="row"
+                      justifyContent="flex-start"
+                      width="fit-content"
+                    >
+                      <Grid item>
+                        <Typography
+                          variant="h5"
+                          sx={{
+                            color: 'rgba(0, 0, 0, 0.6)'
+                          }}
+                        >
+                          Custom Patient Search
+                        </Typography>
+                      </Grid>
+                    </Grid>
+                    <FieldArray name="$or">
+                      {({ push, remove }) => (
+                        <>
+                          {values.$or.map((parameters, index) => {
+                            return (
+                              <FieldGroup
+                                values={parameters}
+                                handleChange={handleChange}
+                                initialCustomSearchValues={
+                                  initialCustomSearchValues
+                                }
+                                fieldGroupIndex={index}
+                                removeFieldGroup={remove}
+                                key={index}
+                                setFieldValue={setFieldValue}
+                              />
+                            )
+                          })}
+
+                          <Grid
+                            item
+                            container
+                            direction="column"
+                            width="100%"
+                            alignItems={'center'}
+                            sx={{ mt: 1 }}
+                          >
+                            <Grid
+                              item
+                              container
+                              direction="row"
+                              width="756px"
+                              justifyContent={'flex-end'}
+                            >
+                              <Button
+                                variant="text"
+                                startIcon={<AddIcon />}
+                                onClick={() => {
+                                  push({
+                                    parameters: [initialCustomSearchValues]
+                                  })
+                                }}
+                                sx={{ fontSize: '13px' }}
+                              >
+                                Add Group
+                              </Button>
```

Comment:
```suggestion
                              <Button
                                variant="text"
                                startIcon={<AddIcon />}
                                onClick={() => {
                                  push({
                                    parameters: [initialCustomSearchValues]
                                  })
                                }}
                                sx={{ fontSize: '13px' }}
                              >
                                Add Group
                              </Button>
```
Could we refactor this code into a seperate component that we could use for "Add field" and "Add group" ?

Reply:
Thanks